### PR TITLE
[FIXES] Release candidate fixes

### PR DESCRIPTION
**Pull request by iceseer**
Signed-off-by: iceseer <iceseer@gmail.com>

### Description of the Change
[RDB] print current DB status on exit
[Utils] ```Internal storage``` implementation. ```Cache``` based on ```internal storage```.
```GetTransactions``` request fixup.
[RDB] memory leaks fixes
[YAC] logical fixes
[OS] remove duplicated txs
[SE] ```repeat``` implementation
[gRPC Client] single thread per peer for batch send process
[Test] ```on_demand_os_client_grpc``` freeze fix

### CODE REVIEW

File: irohad/ametsuchi/impl/in_memory_storage.hpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/**
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/in_memory_storage.hpp:1]: I think we introduced that for the purpose of testing different assumptions. Do we need to keep that? if so, we should write some comment saying that this implementation is for testing purposes only

File: irohad/ametsuchi/impl/rocksdb_common.hpp [hunk 2]
```diff
@@ -464,29 +466,74 @@ namespace iroha::ametsuchi {
...
+
+   private:
+    expected::Result<void, DbError> reinitDB() {
+      assert(db_name_);
+
+      rocksdb::BlockBasedTableOptions table_options;
+      table_options.block_cache =
+          rocksdb::NewLRUCache(1 * 1024 * 1024 * 1024LL);
+      table_options.block_size = 32 * 1024;
+      // table_options.pin_l0_filter_and_index_blocks_in_cache = true;
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/rocksdb_common.hpp:481]: left intentionally?

File: irohad/ordering/impl/batches_cache.cpp [hunk 3]
```diff
@@ -36,6 +39,11 @@ namespace iroha::ordering {
       tx_count_ += batch->transactions().size();
 
     assert(count(batches_) == tx_count_);
+    std::cout << fmt::format(
```
Comment (kamilsa) [inline irohad/ordering/impl/batches_cache.cpp:42]: remove

File: irohad/ordering/impl/batches_cache.cpp [hunk 4]
```diff
@@ -47,6 +55,10 @@ namespace iroha::ordering {
       tx_count_ -= batch->transactions().size();
 
     assert(count(batches_) == tx_count_);
+    std::cout << fmt::format(
```
Comment (kamilsa) [inline irohad/ordering/impl/batches_cache.cpp:58]: remove

**Review by kamilsa — APPROVED** [review]

**Review by kuvaldini — APPROVED** [review]

**Pull request merged**